packages/headless/src/api/platform-client.ts

```diff
@@ -103,6 +97,34 @@ export class PlatformClient {
       return error as PlatformClientCallError;
     }
   }
+
+  private static async preprocessRequest(
+    defaultRequestOptions: PlatformRequestOptions,
+    options: PlatformClientCallOptions
+  ) {
+    const {origin, preprocessRequest, logger, requestMetadata} = options;
+
+    let requestInfo: PlatformRequestOptions;
+
+    try {
+      const processedRequest = await preprocessRequest(
+        defaultRequestOptions,
+        origin,
+        requestMetadata
+      );
+      requestInfo = {
+        ...defaultRequestOptions,
+        ...processedRequest,
+      };
+    } catch (e) {
+      logger.error(
+        e as Error,
+        'Platform request preprocessing failed. Returning default request options.'
+      );
+      requestInfo = {...defaultRequestOptions};
```

Comment:
⚠️ preprocessRequest can taint the defaultRequestOptions then throw I think.
We need to ensures we default on the untainted default I think

Reply:
Fixed and added UT